lesson14/main.js

```diff
@@ -0,0 +1,83 @@
+"use strict";
+const wrapper = document.getElementById("wrapper");
+const lists = document.getElementById("lists");
+const createElBtn = document.getElementById("create-element-btn");
+createElBtn.classList.add("btn");
+const modal = document.getElementById("modal");
+const displayModal = document.getElementById("display-modal");
+const jsonURL = "https://myjson.dit.upm.es/api/bins/j16z";
+const removeModal = () => {
+  modal.remove();
+  wrapper.classList.remove("bg_gray");
+};
+
+const loadingIcon = document.createElement("img");
+const startLoading = () => {
+   loadingIcon.id = "loadingIcon";
+   loadingIcon.src = "img/loading-circle.gif";
+   wrapper.appendChild(loadingIcon);
+};
+
+const endLoading = () => loadingIcon.remove();
+
+displayModal.addEventListener("click", function () {
+   modal.style.display = "block";
+   wrapper.classList.add("bg_gray");
+   displayModal.remove();
+});
+
+createElBtn.addEventListener("click", () => {
+   const inputNumber = document.getElementById("number");
+   if (!inputNumber.value === false) {
+      console.log(inputNumber.value);
+      init();
+   } else {
+      alert("数字が未入力です。");
+   }
+});
+
+async function init() {
+   createElBtn.remove();
+   removeModal();
+   startLoading();
+   try {
+      const result = await fetchData();
+      createListElements(result);
+   } catch (e) {
+      const div = document.createElement("div");
+      lists.appendChild(div);
+      div.textContent = ` エラーが発生しました：${e}`;
+      console.log(e);
+   } finally {
+      endLoading();
+      displayModal.remove();
+   }
+}
```

Comment:
IMO
I felt like there was a lot of unwanted processing mixed in.

```suggestion
async function init() {
  removeModal();
  startLoading();
  try {
    const result = await fetchData();
    createListElements(result);
  } catch (e) {
    const div = document.createElement("div");
    ul.appendChild(div);
    div.textContent = ` エラーが発生しました：${e}`;
    console.log(e);
  } finally {
    endLoading();
  }
}
```

Reply:
Thank you for your review.
I deleted that unwanted processing. 
I completely forgot to exclude them.
79ea4ff